### PR TITLE
Feature and fix to finish the form elements

### PR DESCRIPTION
**Pull request by JamesNimlos**
* feat(cactus-web): allow refs on TextInput for focus control
* fix(cactus-web): stabilize time on large year changes in DateInput

### CODE REVIEW

File: modules/cactus-web/src/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -47,15 +47,17 @@ const displayStatus = (props: TextInputProps) => {
...
-const TextInputBase = (props: TextInputProps) => {
-  const { className, ...rest } = omitMargins(props)
-
-  return (
-    <div className={className}>
-      <Input {...rest} />
-    </div>
-  )
-}
+const TextInputBase = React.forwardRef<HTMLInputElement, TextInputProps>(
```
Comment (Dhalton) [inline modules/cactus-web/src/TextInput/TextInput.tsx]: Looks good just curious why you made this a `forwardRef` component?

Comment (Dhalton) [reply modules/cactus-web/src/TextInput/TextInput.tsx]: I mean, I can see from your commit message that it was done so you can control the focus, I just don't see that being done anywhere

Comment (JamesNimlos) [reply modules/cactus-web/src/TextInput/TextInput.tsx]: It's to use it with an [autosuggest library](https://github.com/moroshko/react-autosuggest) in EBPP

Comment (JamesNimlos) [reply modules/cactus-web/src/TextInput/TextInput.tsx]: I'll point it out when I make the PR, but it was using a normal `input[type=text]` before so not even a semantic-ui component

**Pull request merged**